yt/visualization/plot_window.py

```diff
@@ -2293,14 +2294,63 @@ def _create_axes(self, axrect):
         self.axes = self.figure.add_axes(axrect, projection=self._projection)
 
 
-def SlicePlot(ds, normal=None, fields=None, axis=None, *args, **kwargs):
+def _sanitize_normal_vector(ds, normal) -> Union[str, np.ndarray]:
+    """Return the name of a cartesian axis whener possible,
+    or a 3-element 1D ndarray of float64 in any other valid case.
+    Fail with a descriptive error message otherwise.
+    """
+    axis_names = ds.coordinates.axis_order
+
+    if isinstance(normal, str):
+        if normal not in axis_names:
+            raise ValueError(
+                f"'{normal}' is not a valid axis name. " "Expected either ."
+            )
+        return normal
+
+    if isinstance(normal, (int, np.integer)):
+        if normal not in (0, 1, 2):
+            raise ValueError(
+                f"{normal} is not a valid axis identifier. Expected either 0, 1, or 2."
+            )
+        return axis_names[normal]
+
+    if not is_sequence(normal):
+        raise TypeError(
+            f"{normal} is not a valid normal vector identifier. "
+            "Expected a string, integer or sequence of 3 floats."
+        )
+
+    if len(normal) != 3:
+        raise ValueError(
+            f"{normal} with length {len(normal)} is not a valid normal vector. "
+            "Expected a 3-element sequence."
+        )
+
+    try:
+        retv = np.array(normal, dtype="float64")
+        if retv.shape != (3,):
+            raise ValueError(f"{normal} is incorrectly shape.")
+    except ValueError as exc:
+        raise TypeError(f"{normal} is not a valid normal vector.") from exc
```

Comment:
Is it really a `TypeError`?

Reply:
Arguably. Quoting the standard lib: (https://docs.python.org/3/library/exceptions.html#TypeError):

>Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError, but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a ValueError.

So here, numpy is raising `ValueError` when it fails to cast values to floats, but really it seems to me that it's a `TypeError` by definition

Reply:
:shrug: